src/HaskellCI/Sh.hs

```diff
@@ -97,7 +98,7 @@ instance Applicative ShM where
     pure x = ShM $ \shs -> Right (shs, x)
     (<*>) = ap
 
-instance Monad ShM where
+instance Monad.Monad ShM where
```

Comment:
hmm?

Reply:
Oh, if we use `base-compat-0.11` only, I'd remove the `fail` from `Monad` definition completely.

Reply:
> I'd remove the fail from `Monad` definition completely.

This would make `fail` have different semantics on old versions of `base`, as it would default to invoking `error` instead of `ShM $ \_ -> Left err`. Are you OK with this?

Reply:
`haskell-ci` is `base-compat >=0.10` so `fail` exported from `base-compat-0.11` is always `Control.Monad.Fail.fail`, isn't it?

Reply:
> `haskell-ci` is `base >=4.10` so `fail` exported from `base-compat-0.11` is always `Control.Monad.Fail.fail`, isn't it?

It is, but only assuming you use `Prelude.Compat` absolutely everywhere and you never make use of `MonadFailDesugaring` (which will use `Monad.fail` on pre-8.8 GHCs).

Reply:
`MonadFailDesugaring` :(((

Let's remove the whole `MonadFail` instance. We have `MonadErr` for errors anyway. Then `MonadFailDesugaring` stuff will fail on 8.8.1 - and seems we don't use it.

Reply:
I've removed the `{Monad,MonadFail}.fail` implementation for `ShM`.